src/Windows/Avalonia.Win32/FramebufferManager.cs

```diff
@@ -1,41 +1,193 @@
 ﻿using System;
+using System.Threading;
 using Avalonia.Controls.Platform.Surfaces;
 using Avalonia.Platform;
 using Avalonia.Win32.Interop;
 
+#nullable enable
+
 namespace Avalonia.Win32
 {
-    class FramebufferManager : IFramebufferPlatformSurface, IDisposable
+    internal class FramebufferManager : IFramebufferPlatformSurface, IDisposable
     {
+        private const int _bytesPerPixel = 4;
+        private const PixelFormat _format = PixelFormat.Bgra8888;
+
         private readonly IntPtr _hwnd;
-        private WindowFramebuffer _fb;
+        private readonly object _lock;
+        private readonly Action _onDisposeAction;
+
+        private FramebufferData? _framebufferData;
 
         public FramebufferManager(IntPtr hwnd)
         {
             _hwnd = hwnd;
+            _lock = new object();
+            _onDisposeAction = DrawAndUnlock;
         }
 
         public ILockedFramebuffer Lock()
         {
-            UnmanagedMethods.GetClientRect(_hwnd, out var rc);
+            var lockAcquired = false;
 
-            var width = Math.Max(1, rc.right - rc.left);
-            var height = Math.Max(1, rc.bottom - rc.top);
+            Monitor.Enter(_lock, ref lockAcquired);
```

Comment:
With this pattern Monitor.Enter is supposed to be inside `try` block.  We should probably just get rid of `lockAcquired` here